strax/context.py

```diff
@@ -394,9 +394,34 @@ def _set_plugin_config(self, p, run_id, tolerant=True):
             except strax.InvalidConfiguration:
                 if not tolerant:
                     raise
+
         p.config = {k: v for k, v in config.items()
                     if k in p.takes_config}
 
+        if p.child_ends_with:
+            # This plugin is a child of another plugin and has some different
+            # Checking if the parent plugin is registered, this does not have to be the case
+            # but enables some useful checks.
+            parent_class = self._plugin_class_registry[p.provides[-1]].__bases__[0]
+            is_parent_reg = parent_class in self._plugin_class_registry.values()
+            # options to pass. So update parent config according to child:
+            for k, opt in p.takes_config.items():
+                if k.endswith(p.child_ends_with):
+                    if opt.child_option:
+                        v = config[k]
+                        kparent = k[:-len(p.child_ends_with)]
+
+                        if is_parent_reg:
+                            mes = f'Option {kparent} is not taken by parent plugin.'
+                            assert kparent in parent_class.takes_config.keys(), mes
+
+                        p.config[kparent] = v
+                    else:
+                        warnings.warn(f'You specified plugin {p.__class__.__name__} as a child plugin.'
```

Comment:
Would a ValueError also be appropriate? I imagine that if people do this kind of development they should *not* miss this message to prevent people from writing messy plugins

Reply:
I had the same idea in the beginning and learned afterwards in the hard way that this is not a good choice. Imagine you would like to introduce a new child-only-option. Most likely you will call it with the same ending as the other options. Hence since it is a child-only-option you wont have any corresponding parent which will raise then this error. This is why I changed it in a warning. 
I thought a bit about it, but I was not able to find a more elegant solution for this problem.

Reply:
Hmm yes I see, but, on the other hand, you are changing the option to be a child_only option right so you can also expect someone to change the naming. The thing I worry about is that if someone is making a new plugin and you are not checking it out locally to verify everything works you won't hit this warning whereas if you make it an error you'd instantly fail travis. 

Later you might be stuck with something that throws errors at you all the time but you don't feel like changing it because it would require going back to some low level data.